File: Codes from the AI-assistants/[CLBG] mandelbrot/Basic Prompt - Normal Mode/Mandelbrot-deepseek-basic-normal.py
import sys
import math

def mandelbrot(size, max_iter=1000):
    """Generate Mandelbrot set and output as PBM"""
    sys.stdout.buffer.write(f"P4\n{size} {size}\n".encode())
    
    y_step = 2.0 / size
    x_step = 2.0 / size
    bytes_per_row = (size + 7) // 8
    
    for y in range(size):
        c_im = -1.0 + y * y_step
        byte = 0
        bit_pos = 0
        
        for x in range(size):
            c_re = -1.5 + x * x_step
            c = complex(c_re, c_im)
            
            # Optimization: check main cardioid and period-2 bulb
            q = (c.real - 0.25) ** 2 + c.imag ** 2
            if q * (q + (c.real - 0.25)) <= 0.25 * c.imag ** 2 or \
               (c.real + 1) ** 2 + c.imag ** 2 <= 0.0625:
                iter_count = max_iter
            else:
                # Standard iteration
                z = complex(0, 0)
                iter_count = 0
                while iter_count < max_iter and z.real * z.real + z.imag * z.imag <= 4.0:
                    z = z * z + c
                    iter_count += 1
            
            # Set bit if in Mandelbrot set
            if iter_count == max_iter:
                byte |= (1 << (7 - bit_pos))
            
            bit_pos += 1
            if bit_pos == 8 or x == size - 1:
                sys.stdout.buffer.write(bytes([byte]))
                byte = 0
                bit_pos = 0

if __name__ == "__main__":
    size = int(sys.argv[1]) if len(sys.argv) > 1 else 1000
    mandelbrot(size)

File: Codes from the AI-assistants/[CLBG] mandelbrot/Basic Prompt - Normal Mode/Mandelbrot-deepseek-basic-normal.rs
use std::io::{self, Write};

fn mandelbrot(size: usize, max_iter: usize) -> io::Result<()> {
    let mut stdout = io::stdout();
    
    // Write PBM header
    writeln!(stdout, "P4\n{} {}", size, size)?;
    
    let y_step = 2.0 / size as f64;
    let x_step = 2.0 / size as f64;
    
    for y in 0..size {
        let c_im = -1.0 + y as f64 * y_step;
        let mut byte: u8 = 0;
        let mut bit_pos: u8 = 0;
        
        for x in 0..size {
            let c_re = -1.5 + x as f64 * x_step;
            
            // Check main cardioid and period-2 bulb
            let q = (c_re - 0.25).powi(2) + c_im.powi(2);
            let iter = if q * (q + (c_re - 0.25)) <= 0.25 * c_im.powi(2) ||
                (c_re + 1.0).powi(2) + c_im.powi(2) <= 0.0625 {
                max_iter
            } else {
                // Standard iteration
                let (mut z_re, mut z_im) = (0.0, 0.0);
                let mut iter = 0;
                
                while iter < max_iter && z_re * z_re + z_im * z_im <= 4.0 {
                    let new_re = z_re * z_re - z_im * z_im + c_re;
                    let new_im = 2.0 * z_re * z_im + c_im;
                    z_re = new_re;
                    z_im = new_im;
                    iter += 1;
                }
                iter
            };
            
            // Set bit if in Mandelbrot set
            if iter == max_iter {
                byte |= 1 << (7 - bit_pos);
            }
            
            bit_pos += 1;
            if bit_pos == 8 || x == size - 1 {
                stdout.write_all(&[byte])?;
                byte = 0;
                bit_pos = 0;
            }
        }
    }
    Ok(())
}

fn main() -> io::Result<()> {
    let size = std::env::args()
        .nth(1)
        .and_then(|s| s.parse().ok())
        .unwrap_or(1000);
    let max_iter = std::env::args()
        .nth(2)
        .and_then(|s| s.parse().ok())
        .unwrap_or(1000);
    
    mandelbrot(size, max_iter)
}